### ISSUE TITLE
Piro compilation error due to missing Optipack_Config.h file 

### ISSUE THREAD

**Issue by ikalash** [trilinos/Trilinos#4904]
Trilinos failed to compile in the Albany nightlies due to the following error in Piro:

```
repos/Trilinos/packages/piro/src/Piro_PerformAnalysis.hpp:53:10: fatal error: OptiPack_Config.h: No such file or directory
```

Did OptiPack_Config.h get removed?  Is it still needed?  Here is the more detailed error message:
http://cdash.sandia.gov/CDash-2-3-0/viewBuildError.php?buildid=83699

@trilinos/piro

**Comment by mhoemmen** [comment]
Likely related to this recently merged PR: https://github.com/trilinos/Trilinos/pull/4864

**Comment by bartlettroscoe** [comment]
CC: @kddevin 

This has broken all of the ATDM Trilinos builds as well.  Here is the bad include:

* https://github.com/trilinos/Trilinos/blob/f59aedf03476284d97c91c4d4e0861e01aa6842e/packages/piro/src/Piro_PerformAnalysis.hpp#L53

The problem is that if OptiPack is not enabled, then this triggers the include of a header file that can't be found.

This is trivial to fix.  I will put in a PR.

**Comment by bartlettroscoe** [comment]
CC: @kddevin, @ikalash 

Turns out this is not so trivial to fix.  I pushed the branch:

```
To github.com:bartlettroscoe/Trilinos.git
 * [new branch]      4904-fix-bad-optipack-include -> 4904-fix-bad-optipack-include
```

which gets the build to work but we get failing tests:

```
85% tests passed, 2 tests failed out of 13

Subproject Time Summary:
Piro    =  12.46 sec*proc (13 tests)

Total Test time (real) =   1.72 sec

The following tests FAILED:
	  6 - Piro_AnalysisDriver_MPI_4 (Failed)
	  7 - Piro_AnalysisDriverTpetra_MPI_4 (Failed)
Errors while running CTest
```

because the input files used in the tests 

It does not looks like PR #4864 was not tested with OptiPack disabled and everything else enabled and deprecated code not hidden.

I think the best course of action is to revert PR #4864 for now and fix the problems offline and then rebase the branch and create a new PR once all of these issues have been fixed.

I will submit a revert PR.

**Comment by bartlettroscoe** [comment]
CC: @kddevin, @ikalash 

FYI: I created the revert PR #4908 and already merged that branch to the 'atdm-nightly' branch in the commit https://github.com/trilinos/Trilinos/commit/03830e01e13552e1885a505849c5eb78876874ec so the ATDM Trilinos builds will be fixed even if PR #4908 can't get merged today (due to Trilinos PR backups and random failures).

@kddevin, to reproduce the failures, just take whatever configure script you had and set:

```
-DTrilinos_ENABLE_OptiPack=OFF
```

and that should show the errors.  You might want to start with my branch:

```
To github.com:bartlettroscoe/Trilinos.git
 * [new branch]      4904-fix-bad-optipack-include -> 4904-fix-bad-optipack-include
```

since it fixes all of the build issues.  Someone would just want to fix the runtime issues.  Likely Piro developers should be involved in this some since this is there dependency on OptiPack.

When you think you have something working please rebase your branch and create a new PR.  If you need help with that, let me know.

**Comment by kddevin** [comment]
@bartlettroscoe can you point me to instructions for running one of the ATDM tests that failed for this issue?  Thanks.

**Comment by bartlettroscoe** [comment]
@kddevin asked:

> can you point me to instructions for running one of the ATDM tests that failed for this issue? 

You should be able to reproduce this on any of the ATDM machines and with any of the ATDM Trilinos builds as described at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#quick-start

To make this concrete, get on any SNL COE RHEL6 or RHEL7 machine that has the SEMS env, checkout your Trilinos branch, and do, for example:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh  gnu-7.2.0-openmp-release-debug

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_Piro=ON \
  $TRILINOS_DIR

$ make NP=16  # Uses ninja -j16

$ ctest -j16 
```

Or log onto any system and copy and paste the build name for any build you see at:

* https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2019-04-17&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-

**Comment by kddevin** [comment]
Thanks for the instructions, @bartlettroscoe .  I'll do it this morning.

**Comment by bartlettroscoe** [comment]
Technically this was fixed with the revert PR #4908 and we can see these errors are gone from at least one ATDM Trilinos build [here](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4910163).  (The errors in other builds are due to unrelated issue #4929.)

Going to close this for now.  The next PR attempt #4957 should bring this change back without errors
